test/pkg.jl

```diff
@@ -398,4 +398,73 @@ temp_pkg_dir() do
         @test contains(msg, "$package is not an installed package")
         @test !contains(msg, "signal (15)")
     end
+
+    # partial Pkg.update
+    begin
+        nothingtodomsg = "INFO: No packages to install, update or remove\n"
+
+        ret, out, err = @grab_outputs begin
+            Pkg.rm("Example")
+            Pkg.add("Example")
+        end
+        @test ret == nothing && out == ""
+        @test contains(err, "INFO: Installing Example v")
+
+        ret, out, err = @grab_outputs Pkg.update("Example")
+        @test ret == nothing && out == ""
+        @test contains(err, nothingtodomsg)
+
+        ret, out, err = @grab_outputs begin
+            Pkg.rm("Example")
+            Pkg.add("Example", v"0", v"0.4.1-")
```

Comment:
You could use e.g. 3 to avoid a specific version that you know is buggy, or doesn't work for you for some reason: `Pkg.add("Example", v"0", v"0.4.1", v"0.4.2")` means `[0,0.4.1)∪[0.4.2,∞)`, so it will skip version 0.4.1.